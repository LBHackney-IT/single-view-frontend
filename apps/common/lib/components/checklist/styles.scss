@import "~lbh-frontend/lbh/base";

.mtfh-checklist {
  list-style: none;

  &__tick-icon {
    stroke: lbh-colour("lbh-a01");
  }

  &__cross-icon {
    stroke: lbh-colour("lbh-f02");
  }

  li {
    display: flex;
    align-items: center;
    margin-top: 1em;

    svg {
      margin-top: 0;
      margin-right: 27px;
      height: 21px;
    }

    &:first-child {
      margin-top: 0;
    }
  }
}
