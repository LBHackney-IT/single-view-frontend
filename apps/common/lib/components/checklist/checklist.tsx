import React from "react";

import "./styles.scss";

const CrossIcon = () => {
  return (
    <svg
      width="28"
      height="28"
      viewBox="0 0 28 28"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 5L23 23M23 5L5 23"
        stroke="inherit"
        strokeWidth="6"
        strokeLinecap="square"
      />
    </svg>
  );
};

const TickIcon = () => {
  return (
    <svg
      width="31"
      height="24"
      viewBox="0 0 31 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M26 5L12 19L5 12"
        stroke="inherit"
        strokeWidth="6"
        strokeLinecap="square"
      />
    </svg>
  );
};

interface ChecklistProps {
  items: {
    label: string;
    success: boolean;
  }[];
}

export const Checklist = ({ items }: ChecklistProps) => {
  return (
    <ul className="mtfh-checklist">
      {items.map((item, index) => (
        <li key={index}>
          <span className={`mtfh-checklist__${item.success ? "tick" : "cross"}-icon`}>
            {item.success ? <TickIcon /> : <CrossIcon />}
          </span>
          {item.label}
        </li>
      ))}
    </ul>
  );
};
