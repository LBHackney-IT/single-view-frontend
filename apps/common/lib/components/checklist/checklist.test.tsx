import React from "react";

import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { Checklist } from "./checklist";

test("it renders correctly", () => {
  const mockItems = [
    {
      label: "Success",
      success: true,
    },
    {
      label: "Fail",
      success: false,
    },
  ];
  const { container } = render(<Checklist items={mockItems} />);

  expect(screen.getByText("Success")).toBeInTheDocument();
  expect(screen.getByText("Fail")).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
