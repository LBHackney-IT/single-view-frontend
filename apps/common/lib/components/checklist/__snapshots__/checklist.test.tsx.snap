// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <ul
    class="mtfh-checklist"
  >
    <li>
      <span
        class="mtfh-checklist__tick-icon"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 31 24"
          width="31"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M26 5L12 19L5 12"
            stroke="inherit"
            stroke-linecap="square"
            stroke-width="6"
          />
        </svg>
      </span>
      Success
    </li>
    <li>
      <span
        class="mtfh-checklist__cross-icon"
      >
        <svg
          fill="none"
          height="28"
          viewBox="0 0 28 28"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 5L23 23M23 5L5 23"
            stroke="inherit"
            stroke-linecap="square"
            stroke-width="6"
          />
        </svg>
      </span>
      Fail
    </li>
  </ul>
</div>
`;
